app/models/course/experience_points_record.rb

```diff
@@ -54,4 +55,8 @@ def level_after_update
     current_exp = course_user.experience_points
     course_user.course.level_for(current_exp + exp_changed)
   end
+
+  def set_awarded_at
+    self.awarded_at ||= Time.zone.now
```

Comment:
@weiqingtoh Sorry >< one more thing, I think we only want to set this for `new_record` right ?  Once the record is saved in db, this callback should not be called.  Otherwise, the awarded_at will always be Time.now for those records with nil columns.